resolwe/flow/managers/base.py

```diff
@@ -353,15 +356,29 @@ def _prepare_context(self, data_id, data_dir, runtime_dir, **kwargs):
         }
 
         # extend the settings with whatever the executor wants
-        self.executor.extend_settings(data_id, files)
+        self.executor.extend_settings(data_id, files, secrets)
 
-        # save the settings into the various files in the data dir
+        # save the settings into the various files in the runtime dir
         settings_dict[ExecutorFiles.FILE_LIST_KEY] = list(files.keys())
         for file_name in files:
             file_path = os.path.join(runtime_dir, file_name)
             with open(file_path, 'wt') as json_file:
                 json.dump(files[file_name], json_file, cls=SettingsJSONifier)
 
+        # Save the secrets in the runtime dir, with permissions to prevent listing the given directory.
+        secrets_dir = os.path.join(runtime_dir, ExecutorFiles.SECRETS_DIR)
+        os.makedirs(secrets_dir, mode=stat.S_IWUSR | stat.S_IXUSR)
+        for file_name, value in secrets.items():
+            file_path = os.path.join(secrets_dir, file_name)
+
+            # Set umask to 0 to ensure that we set the correct permissions.
+            old_umask = os.umask(0)
+            try:
+                with os.fdopen(os.open(file_path, os.O_WRONLY | os.O_CREAT, 0o600), 'w') as raw_file:
```

Comment:
> Will fdesc be closed automatically or does it have to be closed manually?

Yes, `fdopen` returns a file-like object backed by the file descriptor and since we use its context manager, it is closed after exiting the with block.

Reply:
> Yes, `fdopen` returns a file-like object backed by the file descriptor and since we use its context manager, it is closed after exiting the with block.

Ok, perfect.